File: src/batch/healthcare/dao/nocturia_factors.py
from datetime import date
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column

from dao import HealthcareBase

# 健康管理データベース
# 夜間頻尿要因管理テーブルデータクラス


class NocturiaFactors(HealthcareBase):
    __tablename__ = 'nocturia_factors'

    # pid: Mapped[SmallInteger] = mapped_column(ForeignKey("person.id"), name='pid',  primary_key=True)
    pid: Mapped[int] = mapped_column(name='pid', primary_key=True)
    measurementDay: Mapped[date] = mapped_column(name='measurement_day', primary_key=True)
    midnightToiletVisits: Mapped[int] = mapped_column(name='midnight_toilet_visits')
    hasCoffee: Mapped[bool] = mapped_column(name='has_coffee')
    hasTea: Mapped[bool] = mapped_column(name='has_tea')
    hasAlcohol: Mapped[bool] = mapped_column(name='has_alcohol')
    hasNutritionDrink: Mapped[bool] = mapped_column(name='has_nutrition_drink')
    hasSportsDrink: Mapped[bool] = mapped_column(name='has_sports_drink')
    hasDiuretic: Mapped[bool] = mapped_column(name='has_diuretic')
    takeMedicine: Mapped[bool] = mapped_column(name='take_medicine')
    takeBathing: Mapped[bool] = mapped_column(name='take_bathing')
    conditionMemo: Mapped[str] = mapped_column(name='condition_memo')

    def __repr__(self):
        return f"NocturiaFactors(pid={self.pid!r}, " \
               f"measurementDay={self.measurementDay!r}, " \
               f"midnightToiletVisits={self.midnightToiletVisits!r})" \
               f"hasTea={self.hasTea!r}" \
               f"hasAlcohol={self.hasAlcohol!r}" \
               f"hasNutritionDrink={self.hasNutritionDrink!r}" \
               f"hasSportsDring={self.hasSportsDring!r}" \
               f"hasDiuretic={self.hasDiuretic!r}" \
               f"takeMedicine={self.takeMedicine!r}" \
               f"takeBathing={self.takeBathing!r}" \
               f"conditionMemo={len(self.conditionMemo)!r})"

File: src/java-health-care-example/com/examples/android/healthcare/data/NocturiaFactors.java
package com.examples.android.healthcare.data;

public class NocturiaFactors {
    private final int midnightToiletVisits;
    private final boolean hasCoffee;
    private final boolean hasTea;
    private final boolean hasAlcohol;
    private final boolean hasNutritionDrink;
    private final boolean hasSportsDrink;
    private final boolean hasDiuretic;
    private final boolean takeMedicine;
    private final boolean takeBathing;
    private final String conditionMemo;

    public NocturiaFactors(int midnightToiletVisits,
           boolean hasCoffee, boolean hasTea, boolean hasAlcohol, boolean hasNutritionDrink,
           boolean hasSportsDrink, boolean hasDiuretic, boolean takeMedicine, boolean takeBathing,
           String conditionMemo) {
        this.midnightToiletVisits = midnightToiletVisits;
        this.hasCoffee = hasCoffee;
        this.hasTea = hasTea;
        this.hasAlcohol = hasAlcohol;
        this.hasNutritionDrink = hasNutritionDrink;
        this.hasSportsDrink = hasSportsDrink;
        this.hasDiuretic = hasDiuretic;
        this.takeMedicine = takeMedicine;
        this.takeBathing = takeBathing;
        this.conditionMemo = conditionMemo;
    }

    public int getMidnightToiletVisits() { return midnightToiletVisits; }
    public boolean hasCoffee() { return hasCoffee; }
    public boolean hasTea() { return hasTea; }
    public boolean hasAlcohol() { return hasAlcohol; }
    public boolean hasNutritionDrink() { return hasNutritionDrink; }
    public boolean hasSportsDrink() { return hasSportsDrink; }
    public boolean hasDiuretic() { return hasDiuretic; }
    public boolean isTakeMedicine() { return takeMedicine; }
    public boolean isTakeBathing() { return takeBathing; }
    public String getConditionMemo() { return conditionMemo; }

    @Override
    public String toString() {
        return "NocturiaFactors{" +
                "midnightToiletVisits=" + midnightToiletVisits +
                ", hasCoffee=" + hasCoffee +
                ", hasTea=" + hasTea +
                ", hasAlcohol=" + hasAlcohol +
                ", hasNutritionDrink=" + hasNutritionDrink +
                ", hasSportsDrink=" + hasSportsDrink +
                ", hasDiuretic=" + hasDiuretic +
                ", takeMedicine=" + takeMedicine +
                ", takeBathing=" + takeBathing +
                ", conditionMemo='" + conditionMemo + '\'' +
                '}';
    }
}
